js/compiler/igor-controller.js

```diff
@@ -13,16 +12,24 @@ import { spawn } from '../GMConstructor.js';
 /** @type {Job[]} */
 const jobs = [];
 
+const fs = require("node:fs/promises");
```

Comment:
I know it's a little strange but check out the `init.js` script - in essence since GMEdit plugins aren't in ES6 land properly and don't get to use imports I used some silly workarounds to get that working with a dynamic import. Part of the reason was it also meant the TS language server was happy. Side effect is that it isn't happy with trying to import node things via the `import` syntax, so my workaround was piping them in via the `init` script.

Last I remember mixing `require`s in to do this made the LSP very upset and I'm not a fan of mixing them myself either so would prefer this stick with the current setup - see `GMConstructor.js` for what that looks like too.